hack/build-go.sh

```diff
@@ -71,13 +71,27 @@ for arg in $args; do
     elif [ "${target}" = "install" ]; then
         eval "$(go env)"
         BIN_NAME=$(basename $arg)
-        ARCHBIN=${BIN_NAME}-$(git describe --always --tags)-${GOHOSTOS}-${GOHOSTARCH}
+        ARCH_BASENAME=${BIN_NAME}-$(git describe --always --tags)
+        ARCHBIN=${ARCH_BASENAME}-${GOHOSTOS}-${GOHOSTARCH}
         mkdir -p ${CMD_OUT_DIR}/${BIN_NAME}
         (
             cd $arg
             go vet ./...
             go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCHBIN}
             (cd ${CMD_OUT_DIR}/${BIN_NAME} && ln -sf ${ARCHBIN} ${BIN_NAME})
+
+            # build virtctl binaries for other operating systems
+            if [ "${BIN_NAME}" = "virtctl" ]; then
+                if [[ "${GOHOSTOS}" != "linux" ]] || [[ "${GOHOSTARCH}" != "amd64" ]]; then
+                    GOOS=linux GOARCH=amd64 go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-linux-amd64
```

Comment:
Look like we  don't need these checks at all. We would always build the same artifact, no matter on which hostos or arch we are. Looks like we could just build all all the time and then link the correct one based on the hostos and arch where we are currently running on. How time consuming is building all of them?

Reply:
> link the correct one based on the hostos and arch
I think we always want to link the linux/amd64 binary, because that is used for building docker images?
Other than that that matches my comment from above about always building linux, and for virtctl also darwin and windows.
I just checked build times, and I'm a bit surprised tbh: the linux builds need a few seconds only, but the darwin and windows builds around 50 seconds each (on my macbook). That's a bit slow... I will update this PR nevertheless